### PR TITLE
jackson serialisation

### PR DESCRIPTION
**Pull request by duncte123**
kotlinx.serialisation does not work well with java. By using jackson and intermediate classes this can be resolved easily

### CODE REVIEW

File: src/main/kotlin/dev/arbjerg/lavalink/client/PlayerUpdateBuilder.kt [hunk 1]
```diff
@@ -21,30 +18,43 @@ class PlayerUpdateBuilder internal constructor(private val node: LavalinkNode, p
...
+        if (curTrackUpdate.isPresent()) {
+            this.trackUpdate = curTrackUpdate.value.copy(encoded = Omissible.omitted()).toOmissible()
+        }
+
         return this
     }
 
     override fun setIdentifier(identifier: String?): PlayerUpdateBuilder {
-        this.identifier = identifier.toOmissible()
+        this.trackUpdate = PlayerUpdateTrack(
```
Comment (topi314) [inline src/main/kotlin/dev/arbjerg/lavalink/client/PlayerUpdateBuilder.kt:55]: should you unset `encoded` in here?

File: src/main/kotlin/dev/arbjerg/lavalink/client/PlayerUpdateBuilder.kt [hunk 2]
```diff
@@ -21,30 +18,43 @@ class PlayerUpdateBuilder internal constructor(private val node: LavalinkNode, p
...
-        this.trackUserData = null.toOmissible()
+        this.trackUpdate = PlayerUpdateTrack(
+            encoded = Omissible.of(null),
+        ).toOmissible()
         return this
     }
 
     override fun setEncodedTrack(encodedTrack: String?): PlayerUpdateBuilder {
-        this.encodedTrack = Omissible.of(encodedTrack)
+        this.trackUpdate = PlayerUpdateTrack(
```
Comment (topi314) [inline src/main/kotlin/dev/arbjerg/lavalink/client/PlayerUpdateBuilder.kt:37]: should you unset `identifier`in here?

File: src/main/kotlin/dev/arbjerg/lavalink/client/PlayerUpdateBuilder.kt (hunk 1 above)
Comment (duncte123) [reply src/main/kotlin/dev/arbjerg/lavalink/client/PlayerUpdateBuilder.kt:55]: Would you say that encoded takes priority over identifier or that its order of execution?

Comment (topi314) [reply src/main/kotlin/dev/arbjerg/lavalink/client/PlayerUpdateBuilder.kt:55]: Order of execution I'd say

Comment (duncte123) [reply src/main/kotlin/dev/arbjerg/lavalink/client/PlayerUpdateBuilder.kt:55]: Sounds good to me

**Review by topi314 — APPROVED** [review]

**Pull request merged**